app/controllers/application_controller.rb

```diff
@@ -1,12 +1,12 @@
 class ApplicationController < ActionController::Base
+  include Pagy::Backend 
   helper_method :current_user
 
+  def authenticate_user!
+    redirect_to request.referer || root_path unless current_user
+  end
+
   def current_user
-    if session[:user_id]
-      @cureent_user ||= User.find(session[:user_id])
-    else
-      @cureent_user = nil
-    end
+    @current_user ||= User.find_by_id(session[:user_id]).presence || nil
```

Comment:
change `find_by_id` to `find_by(id: ....)`
also it will return `nil` so remove `|| nil`

Reply:
done